### DISCUSSION TITLE
Update all server variables at once

### DISCUSSION THREAD

**Discussion by OpenSource03** [pterodactyl/panel#4281] [Support] [answered]
Hello there!

I'll need to update all server variables at once. For **example** if I want to change all Minecraft versions to 1.8.8 at once, is there any SQL command or something I could do?

Thanks!

**Comment by trenutoo** [comment, marked as answer]
Most likely this is not relevant anymore, but I'll write it out for future reference to others. You can quite easily change all the variables, but that alone does nothing without triggering a reinstall to apply it.

I'd suggest taking a short MySQL tutorial anywhere on the internet. You'd want to use an [UPDATE statement](https://dev.mysql.com/doc/refman/8.0/en/update.html) with a more [beginner-friendly guide reference here.](https://www.mysqltutorial.org/mysql-update-data.aspx) and SELECT statement to find the data.

## Always take a backup of the database before modifying it when you don't know what you are doing. Typing wrong values can modify a lot of server variables.

Looking at the panel database, you can find the `egg_variables` table and inside it `env_variable` column. We'll have to locate the variable that you're looking for, retrieve the ID and target it in `server_variables` table.

### Step 1

Find the  `egg_id` from the admin panel under nests and search using it. On the admin panel, it would be marked as `ID`

![image](https://user-images.githubusercontent.com/10975908/183286807-fe6d0a3c-ed7a-4bd1-b6a9-38dd5e2bea0b.png)

### Step 2

Type the following to enter the MySQL CLI on the machine hosting your panel.

```sql
mysql -u root
````

### Step 3

Replace `2` with your own egg id from the instructions above and run the query.

`SELECT id,env_variable FROM panel.egg_variables WHERE egg_id = '2';`

An example output would be:

```sql
| id | env_variable   |
+----+----------------+
|  3 | SERVER_JARFILE |
|  4 | MC_VERSION     |
|  5 | BUILD_TYPE     |
|  6 | FORGE_VERSION  |
+----+----------------+
```

Here the MC_VERSION variable id is 4. We can now target it.

### Step 4

```sql
UPDATE panel.server_variables SET variable_value = '1.8.8' WHERE variable_id = '4';
```

The query above would set the value of **all** variables matching the variable_id 4 to 1.8.8.

Once you got it figured out for one egg, repeat it again.

```sql
mysql -u root
SELECT id,env_variable FROM panel.egg_variables WHERE egg_id = '2';
UPDATE panel.server_variables SET variable_value = '1.8.8' WHERE variable_id = '4';
````
